File: data/readable_format/fortran/file699.f90
Program Main
    Implicit None
    Character(len=100) :: scrpath = './test/'
    Character(len=5) :: filext = '.txt'

    ! Setting up testing environment
    Call SetupTestEnvironment(scrpath, filext)
    
    ! Running the delevec subroutine
    Call delevec(scrpath, filext)
    
    ! Testing file deletion
    Call TestFileDeletion(scrpath, filext)

Contains

    Subroutine delevec(scrpath, filext)
        Character(len=*), Intent(In) :: scrpath, filext
        Logical :: exist
        Character(len=100) :: filePath

        ! Check and delete EVECFV file
        filePath = Trim(scrpath)//'EVECFV'//Trim(filext)
        Inquire(File=filePath, Exist=exist)
        If (exist) Then
            Open(Unit=71, File=filePath, Status='Old')
            Close(Unit=71, Status='Delete')
        End If

        ! Check and delete EVECSV file
        filePath = Trim(scrpath)//'EVECSV'//Trim(filext)
        Inquire(File=filePath, Exist=exist)
        If (exist) Then
            Open(Unit=71, File=filePath, Status='Old')
            Close(Unit=71, Status='Delete')
        End If
    End Subroutine delevec

    Subroutine SetupTestEnvironment(scrpath, filext)
        Character(len=*), Intent(In) :: scrpath, filext
        Call Execute_Command_Line('mkdir -p '//Trim(scrpath))
        Call Execute_Command_Line('touch '//Trim(scrpath)//'EVECFV'//Trim(filext))
        Call Execute_Command_Line('touch '//Trim(scrpath)//'EVECSV'//Trim(filext))
    End Subroutine SetupTestEnvironment

    Subroutine TestFileDeletion(scrpath, filext)
        Character(len=*), Intent(In) :: scrpath, filext
        Logical :: exist

        ! Check EVECFV file deletion
        Inquire(File=Trim(scrpath)//'EVECFV'//Trim(filext), Exist=exist)
        If (.Not. exist) Then
            Print *, "Test Passed: EVECFV file deleted."
        Else
            Print *, "Test Failed: EVECFV file not deleted."
        End If

        ! Check EVECSV file deletion
        Inquire(File=Trim(scrpath)//'EVECSV'//Trim(filext), Exist=exist)
        If (.Not. exist) Then
            Print *, "Test Passed: EVECSV file deleted."
        Else
            Print *, "Test Failed: EVECSV file not deleted."
        End If
    End Subroutine TestFileDeletion

End Program Main

File: data/readable_format/cpp/file699.cpp
#include <iostream>
#include <fstream>
#include <cstdio> // For remove()
#include <cassert>

void delevec(const std::string& scrpath, const std::string& filext) {
    std::string filePath1 = scrpath + "EVECFV" + filext;
    std::ifstream if1(filePath1);
    if (if1.good()) {
        if1.close();
        std::remove(filePath1.c_str());
    } else {
        if1.close();
    }

    std::string filePath2 = scrpath + "EVECSV" + filext;
    std::ifstream if2(filePath2);
    if (if2.good()) {
        if2.close();
        std::remove(filePath2.c_str());
    } else {
        if2.close();
    }
}

void createTestFile(const std::string& path) {
    std::ofstream ofs(path);
    ofs.close();
}

void testFileDeletion(const std::string& scrpath, const std::string& filext) {
    // Setup - create files to be deleted
    createTestFile(scrpath + "EVECFV" + filext);
    createTestFile(scrpath + "EVECSV" + filext);
    
    // Test deletion
    delevec(scrpath, filext);
    
    // Assertions
    std::ifstream testFile1(scrpath + "EVECFV" + filext);
    assert(!testFile1.good() && "EVECFV file was not deleted.");
    testFile1.close();

    std::ifstream testFile2(scrpath + "EVECSV" + filext);
    assert(!testFile2.good() && "EVECSV file was not deleted.");
    testFile2.close();

    std::cout << "All tests passed." << std::endl;
}

int main() {
    std::string scrpath = "./test/"; // Ensure this directory exists or create it as part of the setup
    std::string filext = ".txt";

    testFileDeletion(scrpath, filext);

    return 0;
}